vizro-core/src/vizro/actions/_callback_mapping/_callback_mapping_utils.py

```diff
@@ -170,9 +170,9 @@ def _get_action_callback_outputs(action_id: ModelID) -> Dict[str, Output]:
 def _get_export_data_callback_outputs(action_id: ModelID) -> Dict[str, List[State]]:
     """Gets mapping of relevant output target name and Outputs for `export_data` action."""
     action = model_manager[action_id]
-    try:
-        targets = action.function["targets"]  # type: ignore[attr-defined]
-    except KeyError:
+
+    targets = action.function._arguments.get("targets")  # type: ignore[attr-defined]
+    if not targets:
```

Comment:
I would like us to avoid using `_arguments` wherever possible so we might be able to remove it. See https://github.com/mckinsey/vizro/pull/97#discussion_r1347697504.

So please leave this as it was (plus similar change elsewhere) 🙂

Reply:
If we need this to handle the other `False`y values for `targets` then let's do like this:
```
try:
    targets = action.function["targets"]  # type: ignore[attr-defined]
except KeyError:
    targets = None    

if not targets:
    targets = _get_components_with_data(action_id=action_id)
```

A bit ugly I know, but it will become nicer once we've resolved https://github.com/mckinsey/vizro/pull/97#discussion_r1347697504 in future.